### ISSUE TITLE
Automatic entries/reminders to fill in whats-new

### ISSUE THREAD

**Issue by jbusecke** [xgcm/xgcm#220]
I find myself forgetting to fill in new features/fixes etc in `whats-new.rst`. Is there a way to automatically add e.g. the PR title and author with each merged PR or as part of the CI?

cc @andersy005 @dcherian

**Comment by andersy005** [comment]
> Is there a way to automatically add e.g. the PR title and author with each merged PR or as part of the CI?

Not that I am aware of... I personally find the `squash merge` feature to be useful when merging PRs. I then use the master/main branch history to update the whats-new/changelog by hand in the documentation since with squash merge you get a condensed history. In case you just want the changelog/whats-new to be updated along the side the PR changes, you may find the [pull request template](https://docs.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository) option to be useful.

**Comment by dcherian** [comment]
I don't think you can automate it without squash merging: https://stackoverflow.com/questions/3523534/good-ways-to-manage-a-changelog-using-git

**Comment by jbusecke** [comment]
Thanks this is super helpful. I'll look into both options soon.

**Comment by dcherian** [comment]
I thought I had seen something recently but its an emacs/magit thing: https://github.com/magit/magit/pull/3928

**Comment by jbusecke** [comment]
@andersy005 I think you were still looking into this? Ill reopen and put an 'inprogress' label on.

**Comment by andersy005** [comment]
> Is there a way to automatically add e.g. the PR title and author with each merged PR or as part of the CI?

@jbusecke, I think automating this would **require extra work**. github-activity is able to generate the changelog when given a timeline (e.g. in the example below I generate the changelog since the `v0.5.2` release) but I don't think it can write this to an existing file in a consistent manner. As a user, you would still need to fix the changelog file manually. Is the lack of automation a dealbreaker? 


```bash
$ github-activity xgcm/xgcm --since v0.5.2
```

```
Running search query:
repo:xgcm/xgcm


Found 10 items, which will take 1 pages
Found 51 items, which will take 2 pages
Downloading:: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 51/51 [00:00<00:00, 67.26issues/s]
# v0.5.2...master@{2021-06-24}
```

([full changelog](https://github.com/xgcm/xgcm/compare/v0.5.2...9d1eab55a1d8bef109d59365aee3718d0a090b79))

## Merged PRs

* add interp metric method [#343](https://github.com/xgcm/xgcm/pull/343) ([@jdldeauna](https://github.com/jdldeauna))
* Add configuration for stale issue bot [#341](https://github.com/xgcm/xgcm/pull/341) ([@andersy005](https://github.com/andersy005))
* Leftover typo from #330 [#339](https://github.com/xgcm/xgcm/pull/339) ([@jbusecke](https://github.com/jbusecke))
* adding metrics method [#336](https://github.com/xgcm/xgcm/pull/336) ([@jdldeauna](https://github.com/jdldeauna))
* Prepare v0.5.2 [#332](https://github.com/xgcm/xgcm/pull/332) ([@jbusecke](https://github.com/jbusecke))

## Contributors to this release

([GitHub contributors page for this release](https://github.com/xgcm/xgcm/graphs/contributors?from=2021-05-27&to=2021-06-24&type=c))

[@andersy005](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Aandersy005+updated%3A2021-05-27..2021-06-24&type=Issues) | [@apatlpo](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Aapatlpo+updated%3A2021-05-27..2021-06-24&type=Issues) | [@asiddi24](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Aasiddi24+updated%3A2021-05-27..2021-06-24&type=Issues) | [@dcherian](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Adcherian+updated%3A2021-05-27..2021-06-24&type=Issues) | [@github-actions](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Agithub-actions+updated%3A2021-05-27..2021-06-24&type=Issues) | [@gmacgilchrist](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Agmacgilchrist+updated%3A2021-05-27..2021-06-24&type=Issues) | [@jbusecke](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Ajbusecke+updated%3A2021-05-27..2021-06-24&type=Issues) | [@jdldeauna](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Ajdldeauna+updated%3A2021-05-27..2021-06-24&type=Issues) | [@kthyng](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Akthyng+updated%3A2021-05-27..2021-06-24&type=Issues) | [@miniufo](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Aminiufo+updated%3A2021-05-27..2021-06-24&type=Issues) | [@navidcy](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Anavidcy+updated%3A2021-05-27..2021-06-24&type=Issues) | [@rabernat](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Arabernat+updated%3A2021-05-27..2021-06-24&type=Issues) | [@raphaeldussin](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Araphaeldussin+updated%3A2021-05-27..2021-06-24&type=Issues) | [@rcaneill](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Arcaneill+updated%3A2021-05-27..2021-06-24&type=Issues) | [@Thomas-Moore-Creative](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3AThomas-Moore-Creative+updated%3A2021-05-27..2021-06-24&type=Issues) | [@TomAugspurger](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3ATomAugspurger+updated%3A2021-05-27..2021-06-24&type=Issues) | [@tomchor](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Atomchor+updated%3A2021-05-27..2021-06-24&type=Issues) | [@vrx-](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Avrx-+updated%3A2021-05-27..2021-06-24&type=Issues) | [@wingtorres](https://github.com/search?q=repo%3Axgcm%2Fxgcm+involves%3Awingtorres+updated%3A2021-05-27..2021-06-24&type=Issues)

**Comment by jbusecke** [comment]
This is definitely super nice for the release notes, and having some manual control allows us to combine e.g. related PRs.

I think the desire to automate this is really just that I keep forgetting to write those damn updates to `whats-new.rst`. Is there a way we can 'nudge' authors, or maybe even block the checks if there is no new entry?

**Comment by jbusecke** [comment]
Maybe just giving a preset in rst would already help a lot? I am always confused by the syntax again haha.
